include/mfmg/common/amge.templates.hpp

```diff
@@ -231,6 +265,44 @@ void AMGe<dim, VectorType>::compute_restriction_sparse_matrix(
   restriction_sparse_matrix.compress(dealii::VectorOperation::add);
 
 #if MFMG_DEBUG
+  // Check that the locally_relevant_global_diag is the sum of the agglomerates
+  // diagonal
+  // TODO do not ask user for the locally_relevant_global_diag
+  pos = 0;
+  dealii::LinearAlgebra::distributed::Vector<typename VectorType::value_type>
+      new_global_diag(locally_relevant_global_diag.get_partitioner());
+  for (unsigned int i = 0; i < n_agglomerates; ++i)
+  {
+    unsigned int const n_local_eig = n_local_eigenvectors[i];
+    for (unsigned int k = 0; k < n_local_eig; ++k)
+    {
+      unsigned int const n_elem = eigenvectors[pos].size();
+      if (k == 0)
+      {
+        for (unsigned int j = 0; j < n_elem; ++j)
+        {
+          dealii::types::global_dof_index const global_pos =
+              dof_indices_maps[i][j];
+          new_global_diag[global_pos] += diag_elements[i][j];
+        }
+      }
+      ++pos;
+    }
+  }
+  new_global_diag.compress(dealii::VectorOperation::add);
+  new_global_diag -= locally_relevant_global_diag;
+  ASSERT((new_global_diag.l2_norm() / locally_relevant_global_diag.l2_norm()) <
```

Comment:
You probably don't want `l2` norm here, and want some `max` norm. Ideally, each diagonal element checked individually in relative norm.